### PR TITLE
Migrating helm 2 to 3 and updating some k8s apis

### PR DESCRIPTION
**Pull request by v-fearam**
Only main deploy
It is pending Azure Pipeline update (deploymentCICD.md).
I created a new ticket to tackle deploymentCICD.md, it is complete outdated.

### CODE REVIEW

File: deployment.md [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 - Azure subscription
   > Important: The user initiating the deployment process must have access to the **Microsoft.Authorization/roleAssignments/write** permission. For more information, see [the Container Insights doc](https://docs.microsoft.com/azure/azure-monitor/insights/container-insights-troubleshoot#authorization-error-during-onboarding-or-update-operation)
-- [Azure CLI 2.0](https://docs.microsoft.com/cli/azure/install-azure-cli)
+- [Azure CLI 2.0.49 or later](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli)
```
Comment (neilpeterson) [inline deployment.md:7]: Can you remove `/en-us` so that the proper local is returned?

File: deployment.md [hunk 2]
```diff
@@ -168,7 +171,7 @@ helm repo add aad-pod-identity https://raw.githubusercontent.com/Azure/aad-pod-i
 
 helm repo update
 
-helm install aad-pod-identity/aad-pod-identity --set=installCRDs=true --set nmi.allowNetworkPluginKubenet=true --name aad-pod-identity --namespace kube-system --version 3.0.3
+helm install aad-pod-identity aad-pod-identity/aad-pod-identity --set installCRDs=true --set nmi.allowNetworkPluginKubenet=true  --namespace kube-system --version 3.0.3
```
Comment (neilpeterson) [inline deployment.md:174]: I think that we can remove the version on this and / or update to 4.0.0. I had added the version to work around compatibility issue with Helm 2.

https://github.com/Azure/aad-pod-identity/blob/master/charts/aad-pod-identity/README.md#helm-chart-and-aad-pod-identity-versions

File: deployment.md (hunk 1 above)
Comment (v-fearam) [reply deployment.md]: done

File: deployment.md (hunk 2 above)
Comment (v-fearam) [reply deployment.md]: I put 4.0.0 in order to avoid breaking unexpectedly . Good point, thanks

**Review by neilpeterson — APPROVED** [review]

**Pull request merged**